### PR TITLE
Feat/typescript init errors fix

### PR DESCRIPTION
**Pull request by baierjak**
IT IS ALL ABOUT THE TYPESCRIPT!

It is a bigger one, but most of changes are renamed file extensions from .js to .ts

new babel preset for building Typescript
new package.json scripts for Typescript
tsconfig, jest.config and tslint config added

All files has tslint disable and also typescript disabled (nocheck) because of a lot of files
I will handle them separately. I have started with one typescript Interface in constanst/index.js

### CODE REVIEW

**Review by Tomas2D — APPROVED** [review]

File: src/utilities/apiRequestType/apiRequestType.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// @ts-nocheck
+/* tslint:disable */
+import { RequestTypeParams } from '../../constants';
```
Comment (horaklukas) [inline src/utilities/apiRequestType/apiRequestType.ts:3]: `RequestTypeParams` is not related to constants anyhow, I would rather put it just inside this file

File: tsconfig.types.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "moduleResolution": "node",               /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+        // Generate corresponding .d.ts file.
+        "declaration": true,
+        "resolveJsonModule": true,
+        "allowSyntheticDefaultImports": true,     /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+        "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
+        "skipLibCheck": true,                     /* Skip type checking of declaration files. */
+        "forceConsistentCasingInFileNames": true, /* Disallow inconsistently-cased references to the same file. */
+        "jsx": "react",
+        "outDir": "es"
```
Comment (horaklukas) [inline tsconfig.types.json:23]: are `d.ts` emitted also into the `lib` or just `es` dir?

**Review by horaklukas — APPROVED** [review]

File: tsconfig.types.json (hunk 2 above)
Comment (baierjak) [reply tsconfig.types.json:23]: I simply dont know - how it should be?
Do I need `d.ts` files here at all?
I am a newbie at Typescript but I dont see any purpose for use them here.

File: src/utilities/apiRequestType/apiRequestType.ts (hunk 1 above)
Comment (baierjak) [reply src/utilities/apiRequestType/apiRequestType.ts:3]: yes thats right

File: tsconfig.types.json (hunk 2 above)
Comment (baierjak) [reply tsconfig.types.json:23]: For now the `d.ts` files are generated only in `es` dir

image

Comment (baierjak) [reply tsconfig.types.json:23]: any additional answer guys here? :)

Comment (horaklukas) [reply tsconfig.types.json:23]: I checked other packages and d.ts files are also emitted only into the one file so let's keep it until we find out it's wrong 🙂

File: package.json [hunk 3]
```diff
@@ -23,20 +33,25 @@
...
         "reselect": "4.x"
     },
     "devDependencies": {
         "@babel/cli": "7.x",
         "@babel/core": "7.x",
+        "@babel/plugin-proposal-class-properties": "^7.8.3",
+        "@babel/plugin-proposal-nullish-coalescing-operator": "^7.8.3",
         "@babel/plugin-proposal-object-rest-spread": "7.x",
         "@babel/plugin-transform-runtime": "7.x",
         "@babel/preset-env": "7.x",
```
Comment (jancama2) [inline package.json:49]: Not used here?

File: package.json [hunk 4]
```diff
@@ -23,20 +33,25 @@
...
         "@babel/preset-modules": "0.x",
```
Comment (jancama2) [inline package.json:50]: Not used here?

File: package.json [hunk 5]
```diff
@@ -54,6 +70,10 @@
         "prettier": "1.x",
         "prettier-config-ackee": "0.x",
         "reselect": "4.x",
+        "ts-jest": "25.x",
+        "tslint-config-ackee": "^0.3.0",
```
Comment (jancama2) [inline package.json:74]: TSlint is deprecated in favor of ESlint so pls upgrade all the eslint packages and use it :)

File: package.json [hunk 6]
```diff
@@ -46,6 +61,7 @@
         "eslint-import-resolver-webpack": "0.x",
```
Comment (jancama2) [inline package.json:61]: No webpack used in the project

File: package.json [hunk 7]
```diff
@@ -54,6 +70,10 @@
         "prettier": "1.x",
         "prettier-config-ackee": "0.x",
         "reselect": "4.x",
+        "ts-jest": "25.x",
+        "tslint-config-ackee": "^0.3.0",
+        "tslint-config-prettier": "1.x",
+        "typescript": "3.x",
         "webpack": "4.x"
```
Comment (jancama2) [inline package.json:77]: No webpack used in the project

File: package.json [hunk 8]
```diff
@@ -6,15 +6,25 @@
...
+        "test": "BABEL_ENV=test jest",
+        "push": "yarn build && yalc push",
+        "type-check": "tsc --noEmit",
+        "type-check:watch": "yarn type-check -- --watch",
+        "start": "yarn build && onchange 'src/**/*.ts' -- yarn rebuild",
+        "rebuild": "yarn clean && yarn build:es && yalc push",
+        "changelog": "gitmoji-changelog",
+        "version": "yarn changelog && code --wait CHANGELOG.md && git add CHANGELOG.md",
+        "release": "yarn version",
+        "prettier": "prettier --config ./prettier.config.ts --write './src/**/*.ts'",
```
Comment (jancama2) [inline package.json:26]: Do we need a command for it? It is done by husky.

File: webpack.config.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 const { webpackAliases } = require('./config/aliases');
 
 module.exports = {
```
Comment (jancama2) [inline webpack.config.js:3]: This file is unnecessary as we don't use webpack in the project.

**Review by jancama2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (baierjak) [reply package.json:74]: yes, I will let this as it is now - and after team meeting this week I will change it - when i will have more information about it :)

File: package.json (hunk 8 above)
Comment (baierjak) [reply package.json:26]: maybe not - husky even doesnt use this command :) cool

File: webpack.config.js (hunk 9 above)
Comment (baierjak) [reply webpack.config.js:3]: 👍

File: package.json (hunk 4 above)
Comment (baierjak) [reply package.json:50]: babel uses this, cant delete it :)

File: package.json (hunk 3 above)
Comment (baierjak) [reply package.json:49]: babel uses this, cant delete it :) for browslist ... 
https://babeljs.io/docs/en/babel-preset-env

File: package.json (hunk 6 above)
Comment (baierjak) [reply package.json:61]: yes

File: package.json (hunk 7 above)
Comment (baierjak) [reply package.json:77]: yes

File: package.json (hunk 3 above)
Comment (jancama2) [reply package.json:49]: I cant see it anywhere used tho :)

Comment (jancama2) [reply package.json:49]: right, missed it in the config, srry

File: package.json (hunk 4 above)
Comment (jancama2) [reply package.json:50]: right, missed it in the config, srry

**Pull request merged**